labs/multiverse/src/widgets/room_view/mod.rs

```diff
@@ -565,6 +579,17 @@ impl RoomView {
         }
     }
 
+    fn get_selected_event(&self) -> Option<Arc<TimelineItem>> {
+        let selected = self.timeline_list.selected()?;
+        let selected_room = self.room_id()?;
+
+        let timelines = self.timelines.lock();
+        let current_timeline = timelines.get(selected_room)?;
+        let items = current_timeline.items.lock();
+
+        items.get(selected).cloned()
+    }
```

Comment:
I think that, to work nicely for threads, this should be:

```suggestion
    fn get_selected_event(&self) -> Option<Arc<TimelineItem>> {
        let selected = self.timeline_list.selected()?;
        let items = self.get_selected_timeline_items()?;
        items.get(selected).cloned()
    }
```

But the one caller of this function also calls `get_selected_timeline_items()`, so there's likely a way to avoid this function (that would be now 3 lines), and have the caller get the selected item from `items` (at line 648) immediately.

Reply:
> But the one caller of this function also calls get_selected_timeline_items()

No, it's not doing that anymore.